### ISSUE TITLE
If there is no ‘ip=xx.xx.xx.xx    localhost ’  in the /etc/hosts  file, the VXLAN  service can not work properly

### ISSUE THREAD

**Issue by xiaoluhong** [rancher/rancher#10188]
**Rancher versions:**
rancher/server:v1.6.11-rc4
rancher/agent:v1.2.7-rc2 

**Infrastructure Stack versions:**
rancher/etcd:v3.0.17-4
rancher/etcd:holder
rancher/k8s:v1.8.1-rancher2
rancher/healthcheck:v0.3.3
rancher/k8s:v1.8.1-rancher2
rancher/metadata:v0.9.4
rancher/etcd:v3.0.17-4
rancher/net:v0.13.1
rancher/net:v0.13.1
rancher/network-manager:v0.7.13
rancher/agent:v1.2.7-rc2

kubernetes (if applicable):
v1.8.1-rancher2
**Docker version: (`docker version`,`docker info` preferred)**
root@node2:~# docker info 
Containers: 12
 Running: 11
 Paused: 0
 Stopped: 1
Images: 15
Server Version: 1.12.3
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 87
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-87-generic
Operating System: Ubuntu 16.04.3 LTS

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
"Ubuntu 16.04.3 LTS"   4.4.0-87-generic 
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
VMware Fusion
**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher
**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Kubernetes
**Steps to Reproduce:**
Hi ALL
Today, test deployment of kubernetes in rancher/server:v1.6.11-rc4, I have three nodes, the hostname is node1, node2, node3, and the configuration of the host IP+hostname on each host in the /etc/hosts file, but no configuration of the host IP+localhost,

**Results:**
```
2017/10/22 下午3:49:04time="2017-10-22T07:49:04Z" level=info msg="Listening on localhost:8111"
2017/10/22 下午3:49:04time="2017-10-22T07:49:04Z" level=error msg="got error while ListenAndServe: listen tcp: lookup localhost on 169.254.169.250:53: no such host"
2017/10/22 下午3:49:04time="2017-10-22T07:49:04Z" level=error msg="error: listen tcp: lookup localhost on 169.254.169.250:53: no such host"
2017/10/22 下午3:51:19+ trap 'exit 1' SIGTERM SIGINT
2017/10/22 下午3:51:19+ '[' false == true ']'
2017/10/22 下午3:51:19+ DEBUG=
2017/10/22 下午3:51:19+ exec rancher-vxlan
2017/10/22 下午3:51:19time="2017-10-22T07:51:19Z" level=info msg="vxlan: Start"
2017/10/22 下午3:51:19time="2017-10-22T07:51:19Z" level=info msg="Listening on localhost:8111"
2017/10/22 下午3:51:19time="2017-10-22T07:51:19Z" level=error msg="got error while ListenAndServe: listen tcp: lookup localhost on 169.254.169.250:53: no such host"
2017/10/22 下午3:51:19time="2017-10-22T07:51:19Z" level=error msg="error: listen tcp: lookup localhost on 169.254.169.250:53: no such host"
```

Do I have to configure hostip+localhost?
thanks.

<img width="974" alt="2" src="https://user-images.githubusercontent.com/20707603/31859693-9c432ba4-b741-11e7-8030-c90bf4bd9c9e.png">

**Comment by leodotcloud** [comment]
@niusmallnan can you take a look at this?

**Comment by niusmallnan** [comment]
```
	app.Flags = []cli.Flag{
		cli.StringFlag{
			Name:   "listen",
			Value:  "localhost:8111",
			EnvVar: "RANCHER_SERVICE_LISTEN_PORT",
		},
```
VXLAN needs to listen `localhost:8111`, but in this case the host cannot resolve `localhost`. The system will automatically find dns nameserver, it will automatically call rancher-dns, then we can get these logs:
```
listen tcp: lookup localhost on 169.254.169.250:53: no such host
```

We can use `127.0.0.1` instead of `localhost` to avoid this problem.
In fact, if the user's OS does not have `localhost` mapping, I guess not only have this problem.

@leodotcloud @yasker

**Comment by hongxiaolu** [comment]
thanks.

**Comment by leodotcloud** [comment]
Fixed in `rancher/net:v0.13.8`
